carrier.go

```diff
@@ -92,29 +115,60 @@ func (c *Client) GetCarrierTypesWithContext(ctx context.Context) (out []*Carrier
 //			},
 //		},
 //	)
+//
+// Users cannot create UPS accounts with this function, must use CreateUpsCarrierAccount. An error will be returned if the user tries to create a UPS account with this function.
 func (c *Client) CreateCarrierAccount(in *CarrierAccount) (out *CarrierAccount, err error) {
 	return c.CreateCarrierAccountWithContext(context.Background(), in)
 }
 
-// CreateCarrierAccountWithContext performs the same operation as
-// CreateCarrierAccount, but allows specifying a context that can interrupt the
-// request.
+// CreateCarrierAccountWithContext performs the same operation as CreateCarrierAccount, but allows specifying a context that can interrupt the request.
+// Users cannot create UPS accounts with this function, must use CreateUpsCarrierAccount. An error will be returned if the user tries to create a UPS account with this function.
 func (c *Client) CreateCarrierAccountWithContext(ctx context.Context, in *CarrierAccount) (out *CarrierAccount, err error) {
-	req := &carrierAccountRequest{CarrierAccount: in}
-	endpoint := c.selectCarrierAccountCreationEndpoint(*in)
+	// Users cannot create UPS accounts with this function, must use CreateUpsCarrierAccount
+	for _, carrier := range getUpsCarrierAccountTypes() {
+		if in.Type == carrier {
+			return nil, newInvalidFunctionError("users must use CreateUpsCarrierAccount to create UPS accounts")
+		}
+	}
+
+	req := &carrierAccountRequest{Data: in}
+	endpoint := c.selectCarrierAccountCreationEndpoint(in.Type)
 	err = c.post(ctx, endpoint, req, &out)
 	return
 }
 
-// ListCarrierAccounts returns a list of all carrier accounts available to the
-// authenticated account.
+// CreateUpsCarrierAccount creates a new UPS carrier account. It can only be used with a production API key.
+// Users cannot create non-UPS accounts with this function, must use CreateCarrierAccount. An error will be returned if the user tries to create a non-UPS account with this function.
+func (c *Client) CreateUpsCarrierAccount(in *UpsCarrierAccountCreationParameters) (out *CarrierAccount, err error) {
+	return c.CreateUpsCarrierAccountWithContext(context.Background(), in)
+}
+
+// CreateUpsCarrierAccountWithContext performs the same operation as CreateUpsCarrierAccount, but allows specifying a context that can interrupt the request.
+// Users cannot create non-UPS accounts with this function, must use CreateCarrierAccount. An error will be returned if the user tries to create a non-UPS account with this function.
+func (c *Client) CreateUpsCarrierAccountWithContext(ctx context.Context, in *UpsCarrierAccountCreationParameters) (out *CarrierAccount, err error) {
+	// Users cannot create non-UPS accounts with this function, must use CreateCarrierAccount
+	isUpsAccount := false
+	for _, carrier := range getUpsCarrierAccountTypes() {
+		if in.Type == carrier {
```

Comment:
question: is there no way in Go to just check that a value exists in a range? Surely there is...

Reply:
There's slices, but that's only Go 1.21+: https://pkg.go.dev/slices#Contains
 
 Sometimes the good ol' loop is the best option.